### PR TITLE
chore: add github actions

### PR DESCRIPTION
**Pull request by QuentinBrosse**

### CODE REVIEW

File: .github/workflows/unit-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#on:
+#  push:
+#    branches: [ v2 ]
+#  pull_request:
+
+on:
+  pull_request:
+
+jobs:
+  ubuntu-unit-test:
```
Comment (Sh4d1) [inline .github/workflows/unit-tests.yaml]:
```suggestion
different-os-unit-test
```

File: .github/workflows/unit-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: [ v2 ]
```
Comment (Sh4d1) [inline .github/workflows/unit-tests.yaml]: Running tests once merged? :thinking:

File: .github/workflows/unit-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+name: Unit Tests
```
Comment (jerome-quere) [inline .github/workflows/unit-tests.yaml]: These are not unit test as it's a mix of unit and integration.

File: .github/workflows/unit-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+on:
+  pull_request:
+
+jobs:
+  os-unit-test:
```
Comment (jerome-quere) [inline .github/workflows/unit-tests.yaml]: same remark here

File: .github/workflows/unit-tests.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      - name: Install Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - name: Build binary
+        run: GOARCH=${{ matrix.arch }} go build ./...
```
Comment (jerome-quere) [inline .github/workflows/unit-tests.yaml]: - Should we build cmd/scw ?
- Shoud we use build script that already do cross compilation ?

**Pull request merged**